### DISCUSSION TITLE
Integrate surfaces from FreeSurfer in the charm segmentation pipeline

### DISCUSSION THREAD

**Discussion by dimst23** [simnibs/simnibs#205] [7.Creating head models] [answered]
### Background

I'm currently encountering challenges in effectively segmenting the data sourced from the `OASIS 1` dataset. The issues I'm facing bear a strong resemblance to the difficulties discussed in [this post](https://github.com/simnibs/simnibs/discussions/131). I've attempted to apply the suggested settings from the post, yet the outcomes remain sub-optimal. 

I am not attaching any images because what you see in the aforementioned post is more or less what I get. The gray matter has a smooth texture with minimal sulcal detail.

### Question
I want to ask  if there's a method to integrate the pre-segmented gray matter and white matter surfaces obtained from FreeSurfer into the charm segmentation pipeline. The `OASIS 1` dataset offers FreeSurfer pre-processed data with high sulcal detail.

### General
Even if it is not the FreeSurfer format specifically, is there a way to define a custom surface of any kind along the pipeline, without affecting the final result?

Thank you for your time!

**Comment by oulap** [comment, marked as answer]
We have actually implemented a way to pass the FS surfaces into charm so that the GM gets cleaned with those surfaces. That will be out in the next release, which is hopefully happening soon.

I'll ping this discussion once we have released.

**Reply by dimst23** [reply]
OK, thank you! 
Is there the code under development, stable enough, which I can try now?

**Reply by dimst23** [reply]
I did not check the commit history when I asked about the code availability. Now I have compiled the latest commit which includes the FS functionality. 

It seems to work better than not using FS. I think that discussion can be considered answered.

**Reply by oulap** [reply]
I also just pushed the changes on Friday to GitHub. But good that you found it already!
